### ISSUE TITLE
v1.16.2 git tag forces user to be on Python 3

### ISSUE THREAD

**Issue by didip** [numpy/numpy#14050]
v1.16.x should maintain Python 2 compatibility, yes? But it currently requires Python 3.

### Reproducing code example:

```
# bash
git checkout -b v1.16.2
python setup.py sdist

Traceback (most recent call last):
  File "setup.py", line 31, in <module>
    raise RuntimeError("Python version >= 3.5 required.")
RuntimeError: Python version >= 3.5 required.
```

**Comment by seberg** [comment]
@didip can you make sure to run `git clean -xdf` and then try again?

**Comment by didip** [comment]
Same problem

```
[~/oss/numpy] (v1.16.2) git clean -xdf
[~/oss/numpy] (v1.16.2) python setup.py sdist
Traceback (most recent call last):
  File "setup.py", line 31, in <module>
    raise RuntimeError("Python version >= 3.5 required.")
RuntimeError: Python version >= 3.5 required.
```

**Comment by didip** [comment]
I saw this SHA `badf2901ea`. Should that be merged in `v1.16.x` branches?

**Comment by seberg** [comment]
Well, you may want to use `git checkout -b v1.16.2 v1.16.2` or something, you just created a new branch based on master as far as I can see. The maintenance branch is `maintenance/1.16.x` though, if you want the 1.16.x development version.
